src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Url/Base64UrlDecoder.cs

```diff
@@ -0,0 +1,794 @@
+
+namespace System.Buffers.Text
+{
+    // AVX2 and Vector128 version based on https://github.com/gfoidl/Base64/blob/5383320e28cac6c7ac6f86502fb05d23a048a21d/source/gfoidl.Base64/Internal/Encodings/Base64UrlEncoding.cs
+
+    public static partial class Base64Url
+    {
+        private const int MaxStackallocThreshold = 256;
+
+        /// <summary>
+        /// Returns the maximum length (in bytes) of the result if you were to decode base 64 encoded text within a byte span of size <paramref name="base64Length"/>.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown when the specified <paramref name="base64Length"/> is less than 0.
+        /// </exception>
+        public static int GetMaxDecodedLength(int base64Length)
+        {
+            ArgumentOutOfRangeException.ThrowIfNegative(base64Length);
+
+            (uint whole, uint remainder) = uint.DivRem((uint)base64Length, 4);
+
+            return (int)(whole * 3 + (remainder > 0 ? remainder - 1 : 0));
+        }
+
+        /// <summary>
+        /// Decode the span of UTF-8 encoded text represented as Base64Url into binary data.
+        /// </summary>
+        /// <param name="source">The input span which contains UTF-8 encoded text in Base64Url that needs to be decoded.</param>
+        /// <param name="destination">The output span which contains the result of the operation, i.e. the decoded binary data.</param>
+        /// <param name="bytesConsumed">The number of input bytes consumed during the operation. This can be used to slice the input for subsequent calls, if necessary.</param>
+        /// <param name="bytesWritten">The number of bytes written into the output span. This can be used to slice the output for subsequent calls, if necessary.</param>
+        /// <param name="isFinalBlock"><see langword="true"/> (default) when the input span contains the entire data to encode.
+        /// Set to <see langword="true"/> when the source buffer contains the entirety of the data to encode.
+        /// Set to <see langword="false"/> if this method is being called in a loop and if more input data may follow.
+        /// At the end of the loop, call this (potentially with an empty source buffer) passing <see langword="true"/>.</param>
+        /// <returns>It returns the OperationStatus enum values:
+        /// - Done - on successful processing of the entire input span
+        /// - DestinationTooSmall - if there is not enough space in the output span to fit the decoded input
+        /// - NeedMoreData - only if <paramref name="isFinalBlock"/> is false and the input is not a multiple of 4
+        /// - InvalidData - if the input contains bytes outside of the expected Base64Url range, or if it contains invalid/more than two padding characters
+        /// or if the input is incomplete (i.e. the remainder of <paramref name="source"/> % 4 is 1) and <paramref name="isFinalBlock"/> is <see langword="true"/>.
+        /// </returns>
```

Comment:
https://github.com/dotnet/dotnet-api-docs/wiki/Returns says that if all of these need to be explained, it should be in a table.  Personally, I don't know that it's useful to explain the OperationStatus values here.

```suggestion
        /// <returns>One of the enumeration values that indicates the success or failure of the operation.</returns>
```